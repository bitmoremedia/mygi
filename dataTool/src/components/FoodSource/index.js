import {connect} from 'react-redux';

import FoodSource from './component';

const mapStateToProps = (state, props) => {
  return {
    foodList: state.foodList,
    dataSources: state.dataSources,
  };
};

export default connect(mapStateToProps)(FoodSource);
