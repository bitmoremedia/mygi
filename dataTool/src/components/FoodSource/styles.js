import styled from 'styled-components';

export const List = styled.ul`
`;

export const ListItem = styled.li`
  padding: 5px;
  margin: 2px;
`;

export const MatchedListItem = styled.li`
  padding: 5px;
  margin: 2px;
  background-color: green;
  color: white
`;
