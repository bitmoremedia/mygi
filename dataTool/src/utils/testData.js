const testData = {
  foodList: {
    'abcd-123': {
      category: 'Beans',
      gi: 35,
      id: 'abcd-123',
      name: 'Kidney Beans',
      sources: {},
    },
    'efgh-456': {
      category: 'Bread',
      gi: 68,
      id: 'efgh-456',
      name: 'White Bread',
      sources: {
        havard: 'havard-white-bread',
      },
    },
  },
  dataSources: {
    havard: {
      name: 'havard',
      title: 'Havard',
      data: {
        'havard-white-bread': {
          gi: 65,
          id: 'havard-white-bread',
          name: 'White Bread',
        },
      },
    },
    mendosa: {
      name: 'mendosa',
      title: 'Mendosa',
      data: {
        'mendosa-white-bread': {
          gi: 77,
          id: 'mendosa-white-bread',
          name: 'White Bread',
        },
      },
    },
  },
  foodId: 'mendosa-white-bread',
  sourceName: 'mendosa',
}

export default testData
