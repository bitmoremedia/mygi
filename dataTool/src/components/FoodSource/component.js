import React, { Component } from 'react';
import _values from 'lodash/values';
import _sortBy from 'lodash/sortBy';
import _forEach from 'lodash/forEach';

import { List, ListItem, MatchedListItem, EditLink } from './styles';

import ToolTip from '../common/ToolTip';
import Divider from '../common/Divider';
import Modal from '../common/Modal';
import FoodFinder from '../FoodFinder';

class FoodSource extends Component {
  constructor(props, context) {
    super(props, context);
    this.state = {
      findFoodModalVisible: false,
      findFoodId: undefined,
    };
    this.openFindFoodModal = this.openFindFoodModal.bind(this);
    this.closeFindFoodModal = this.closeFindFoodModal.bind(this);
  }

  openFindFoodModal(e,id) {
    e.preventDefault();
    this.setState({
      findFoodModalVisible: true,
      findFoodId: id
    });
  }

  closeFindFoodModal(e) {
    e.preventDefault();
    this.setState({
      findFoodModalVisible: false,
      findFoodId: undefined
    });
  }

  render() {
    const { match, foodList, dataSources } = this.props;
    const { findFoodModalVisible, findFoodId } = this.state;

    const sourceName = match.params.id;
    const { title, data } = dataSources[sourceName] || {};
    const sourceDataArray = _sortBy(_values(data), 'name');

    const sourceMatch = {};
    if ( data ){
      _forEach(foodList, (food, foodId)=> {
        const sourceFood = data[food.sources[sourceName]];
        if (food.sources[sourceName] && sourceFood) {
          sourceMatch[sourceFood.id] = {
            ...sourceFood,
            matched: {
              ...food
            }
          }
        }
      })
    }

    const FoodItem = ({ id, name, gi }) => {
      if ( sourceMatch[id] ){
        return (
          <MatchedListItem>
            <ToolTip
              text={`${sourceMatch[id].matched.name} [${sourceMatch[id].matched.gi}]`}
              position={'right'}
            >
              âœ”
            </ToolTip> {name} [{gi}] <EditLink href="" onClick={(e)=>this.openFindFoodModal(e,id)}>Edit</EditLink>
          </MatchedListItem>
        );
      }
      return <ListItem>{name} [{gi}] <a href="" onClick={(e)=>this.openFindFoodModal(e,id)}>Find</a></ListItem>;
    }

    return (
      <div>
        <h2>{title}</h2>
        <Modal visible={findFoodModalVisible} onClose={this.closeFindFoodModal} width={80} height={80} measure={'%'}>
          {
            findFoodModalVisible && <FoodFinder foodId={findFoodId} sourceName={sourceName} />
          }
        </Modal>
        <Divider />
        <List>
          {sourceDataArray.map((food)=><FoodItem key={food.id} {...food}/>)}
        </List>
      </div>
    );
  }
}

export default FoodSource;
