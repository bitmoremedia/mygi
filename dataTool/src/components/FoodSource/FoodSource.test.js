import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import testData from '../../utils/testData'

import FoodSource from './component'

test('component renders without error', () => {
  const wrapper = mount(
    <FoodSource
      match={{
        params: {
          id: 'havard',
        },
      }}
      foodList={testData.foodList}
      dataSources={testData.dataSources}
    />,
  )
  expect(wrapper.exists()).toBe(true)
})

// SNAPSHOT TESTS
test('component renders as expected', () => {
  const wrapper = mount(
    <FoodSource
      match={{
        params: {
          id: 'havard',
        },
      }}
      foodList={testData.foodList}
      dataSources={testData.dataSources}
    />,
  )
  expect(toJson(wrapper)).toMatchSnapshot()
})
