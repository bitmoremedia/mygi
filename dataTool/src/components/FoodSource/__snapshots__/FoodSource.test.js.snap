exports[`test component renders as expected 1`] = `
<FoodSource
  dataSources={
    Object {
      "havard": Object {
        "data": Object {
          "havard-white-bread": Object {
            "gi": 65,
            "id": "havard-white-bread",
            "name": "White Bread",
          },
        },
        "name": "havard",
        "title": "Havard",
      },
      "mendosa": Object {
        "data": Object {
          "mendosa-white-bread": Object {
            "gi": 77,
            "id": "mendosa-white-bread",
            "name": "White Bread",
          },
        },
        "name": "mendosa",
        "title": "Mendosa",
      },
    }
  }
  foodList={
    Object {
      "abcd-123": Object {
        "category": "Beans",
        "gi": 35,
        "id": "abcd-123",
        "name": "Kidney Beans",
        "sources": Object {},
      },
      "efgh-456": Object {
        "category": "Bread",
        "gi": 68,
        "id": "efgh-456",
        "name": "White Bread",
        "sources": Object {
          "havard": "havard-white-bread",
        },
      },
    }
  }
  match={
    Object {
      "params": Object {
        "id": "havard",
      },
    }
  }>
  <div>
    <h2>
      Havard
    </h2>
    <Modal
      height={80}
      measure="%"
      onClose={[Function]}
      visible={false}
      width={80}>
      <Rodal
        animation="zoom"
        className=""
        customMaskStyles={Object {}}
        customStyles={Object {}}
        duration={300}
        height={80}
        measure="%"
        onClose={[Function]}
        showCloseButton={true}
        showMask={true}
        visible={false}
        width={80}>
        <div
          className="rodal rodal-fade-leave "
          onAnimationEnd={[Function]}
          style={
            Object {
              "WebkitAnimationDuration": "300ms",
              "animationDuration": "300ms",
              "display": "none",
            }
          }>
          <div
            className="rodal-mask"
            onClick={[Function]}
            style={Object {}} />
          <Dialog
            animation="zoom"
            animationType="leave"
            className=""
            customMaskStyles={Object {}}
            customStyles={Object {}}
            duration={300}
            height={80}
            measure="%"
            onClose={[Function]}
            showCloseButton={true}
            showMask={true}
            visible={false}
            width={80}>
            <div
              className="rodal-dialog rodal-zoom-leave"
              style={
                Object {
                  "WebkitAnimationDuration": "300ms",
                  "animationDuration": "300ms",
                  "height": "80%",
                  "width": "80%",
                }
              }>
              <span
                className="rodal-close"
                onClick={[Function]} />
              <styled.div>
                <div
                  className="CiQaJ" />
              </styled.div>
            </div>
          </Dialog>
        </div>
      </Rodal>
    </Modal>
    <Component>
      <styled.hr>
        <hr
          className="fTafup" />
      </styled.hr>
    </Component>
    <styled.ul>
      <ul
        className="foqgVb">
        <Component
          gi={65}
          id="havard-white-bread"
          name="White Bread">
          <styled.li>
            <li
              className="cHeXBg">
              <Component
                length="large"
                position="right"
                text="White Bread [68]">
                <styled.span>
                  <span
                    className="kpQwSz">
                    <span
                      data-balloon="White Bread [68]"
                      data-balloon-length="large"
                      data-balloon-pos="right">
                      âœ”
                    </span>
                  </span>
                </styled.span>
              </Component>
               
              White Bread
               [
              65
              ] 
              <styled.a
                href=""
                onClick={[Function]}>
                <a
                  className="iriPU"
                  href=""
                  onClick={[Function]}>
                  Edit
                </a>
              </styled.a>
            </li>
          </styled.li>
        </Component>
      </ul>
    </styled.ul>
  </div>
</FoodSource>
`;
